File: 989. Add to Array-Form of Integer/Solution.py
class Solution:
    def addToArrayForm(self, num: List[int], k: int) -> List[int]:
        res = []
        n, carry, total = len(num), 0, 0
        for i in range(n - 1, -1, -1):
            total = num[i] + carry + k % 10
            res.append(total % 10)
            carry = total // 10
            k //= 10
        k += carry
        while k > 0:
            res.append(k % 10)
            k //= 10
        return reversed(res)

File: 989. Add to Array-Form of Integer/Solution.java
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> res = new ArrayList<>();
        int n = num.length, carry = 0, total, i;
        for (i = 1; i <= n; i++) {
            total = num[n - i] + k % 10 + carry;
            res.add(total % 10);
            carry = total / 10;
            k /= 10;
        }
        k += carry;
        while (k > 0) {
            res.add(k % 10);
            k /= 10;
        }
        Collections.reverse(res);
        return res;
    }
}